appcelerator-archive/amp
# Stack network stage 2

#280

Be able to create custom networks and associate stack services to them in order to establish private communication between services from separate stacks.

test:
- stop amp:  sudo ./swarm stop
- pull images: ./swarm pull
- build amp: glide cc && rm -rf vendor && glide install && make install
- start amp: sudo ./swarm start
- create stack test1: amp stack up test1 -f api/rpc/stack/test_samples/sample-10-1.yml
- create stack test2: amp stack up test2 -f api/rpc/stack/test_samples/sample-10-2.yml
- verify:
  - stack test1 and test2 are up and running using: ./swarm monitor
  - custom network "amp_net" is created, using: docker network ls
  - container test1.pinger is able to ping test2.pinger
    - docker ps | grep test1.pinger  -> get the container id [containerId1]
    - docker exec -it [containerId1] sh
    - inside the container:  ping test2-pinger  -> should return a valid address
  - container test2.pinger is able to ping test1.pinger
    - docker ps | grep test2.pinger  -> get the container id [containerId2]
    - docker exec -it [containerId2] sh
    - inside the container:  ping test1-pinger  -> should return a valid address

Regression tests:
- make test

minor:
- add volumes in service definition and in stack file (tests added in `make test`) #299
- add args in services definition and in stack file (tests added in `make test`)
